freqtrade/misc.py

```diff
@@ -249,3 +251,41 @@ def parse_db_uri_for_logging(uri: str):
         return uri
     pwd = parsed_db_uri.netloc.split(':')[1].split('@')[0]
     return parsed_db_uri.geturl().replace(f':{pwd}@', ':*****@')
+
+
+def dataframe_to_json(dataframe: pandas.DataFrame) -> str:
+    """
+    Serialize a DataFrame for transmission over the wire using JSON
+    :param dataframe: A pandas DataFrame
+    :returns: A JSON string of the pandas DataFrame
+    """
+    return dataframe.to_json(orient='records')
```

Comment:
i wonder why we're using `.to_json(orient='records')` here.
it's one of the most verbose formats, resulting in an array that looks as follows (for an empty (!) ohlcv dataframe)

```
[{"date":1513134000000,"open":272.0,"high":288.0,"low":272.0,"close":276.0,"volume":130.86075},{"date":1513137600000,"open":276.0,"high":288.0,"low":269.99,"close":279.0,"volume":606.4813},{"date":1513141200000,"open":279.0,"high":330.0,"low":279.0,"close":302.0,"volume":383.76674},{"date":1513144800000,"open":302.0,"high":304.11,"low":286.0,"close":291.13,"volume":351.59475},{"date":1513148400000,"open":291.13,"high":296.0,"low":285.0,"close":287.0,"volume":165.1242},
...
```

The column name for each column is repeated once per row ... making the transferred size quite big.
Considering our aim here (serialize a DF, transfer it as json, deserialize it to DF) - i think we should pick a better format here - for example the one we use for the rest api (or some deviation of this).
```
"columns": ["date", "open", "high", "low", "close"],
"values": [[1513134000000,272.0,288.0,272.0,276.0,130.86075],[1513137600000,276.0,288.0,269.99,279.0,606.4813],[1513141200000,279.0,330.0,279.0,302.0,383.76674],[1513144800000,302.0,304.11,286.0,291.13,351.59475],[1513148400000,291.13,296.0,285.0,287.0,165.1242],
...
]
```